bee-protocol/src/tangle/metadata.rs

```diff
@@ -30,13 +34,19 @@ impl TransactionMetadata {
         arrival_timestamp: u64,
         solidification_timestamp: u64,
         confirmation_timestamp: u64,
+        cone_index: Option<MilestoneIndex>,
```

Comment:
what exactly is the `cone_index` compared to the `milestone_index`?

Reply:
Both should do the same, and we should remove `cone_index` if possible. TSA relies on `cone_index` at the moment. I tried to use `milestone_index` with the TSA, but what I have seen so far is that the TSA doesn't work with the `milestone_index` anymore (it did lead to an empty tip pool). Unfortunately I had not the time to investigate this more.

Reply:
@rufsam can you help me again with this? What was the difference again?

Reply:
The `cone_index` and `milestone_index` basically should do the same thing. Once a transaction got confirmed by a milestone, set this field with the index of the milestone. The `milestone_index` field however gets maintained by the `bee-ledger` while the `cone_index` gets maintained by the `bee-protocol`. We need the `cone_index` in following places:

here:
https://github.com/Alex6323/bee-p/blob/samuel-wurts/bee-protocol/src/worker/milestone_cone_updater.rs#L77
Basically if it is set, we know the transaction was part of a previous milestone cone and therefore we can stop the traversal.

and here:
https://github.com/Alex6323/bee-p/blob/samuel-wurts/bee-protocol/src/worker/milestone_cone_updater.rs#L101
If it is set, we know we have already processed this transactions and therefore already also propagated OTRSI/YTRSI values to the future. So we can skip this transaction.

Replacing simply the `cone_index` here with `milestone_index` unfortunately doesn't work since OTRSI/YTRSI would be set independently of `milestone_index` and above algorithms wouldn't work anymore.
 
We can remove the `cone_index` when we move all the logic from the https://github.com/Alex6323/bee-p/blob/samuel-wurts/bee-protocol/src/worker/milestone_cone_updater.rs to the bee-ledger. Basically the `milestone_index` needs to be updated together with OTRSI/YTRSI values.

Reply:
Okay, thanks, Sam. @thibault-martinez Do you have to anything to say about this. Do you think we can merge those indexes at some point?